prefq_documentation.md

```diff
@@ -0,0 +1,96 @@
+
+## Development Stage Server (local)
+
+For a quick first impression - or if you'd like to adjust the server script to your individual needs - it is not necessary to host a remote server. Instead you can simply run the provided server script on localhost. This behavior is specified within the last lines of `server.py`:
+
+```python
+if __name__ == "__main__":
+    before_first_request()
+    app.run(host="localhost", port=5000, debug=True)
+```
+
+### Server
+Start your own local server:
+1. Clone the repository: `git clone https://github.com/timokau/prefq.git`
+2. Install poetry: `curl -sSL https://install.python-poetry.org | python3 -`
+3. Add entry to .bashrc:<br>   `export PATH="/root/.local/bin:$PATH"`<br>   `echo 'export PATH="/root/.local/bin:$PATH"'`<br>   `~/.bashrc`
+4. Start the server: `scripts/run_server.sh`
+
+### Query Client
+Send queries to your local server:
+1. Open a new terminal
+2. *(a)* Send static queries: `scripts/run_static_queries.sh`<br>*(b)* Send dynamic queries: `scripts/run_imitation_example.sh`
+
+Option (a):
+- shows the required http-request structure for sending queries & receiving feedback
+- does not learn new behavior
+
+Option (b):
+- integrates our PrefQ-Server into [imitation](https://pypi.org/project/imitation/), a library based on [gymnasium](https://gymnasium.farama.org/index.html)
+- capable of learning new behavior
+
+### Feedback Client
+Evaluate queries:
+1. Visit the web interface: `http://localhost:5000` (*using your favorite web browser*)
+2. Select your preferred video: use left/right arrow keys or click the corresponding button
+
+Using such a development server holds several eminent advantages
+1. Changes within your project folder will be recognized. As soon as you save changes, the server will reboot automatically and you can instantly test and verify new behavior.
+2. Valuable debug information about incoming http-requests will be provided.
+3. You can develop your own server without being connected to the web.
+
+
+## Production Stage Server (remote)
+
+For remote feedback collection it is also possible to run a development stage server remotely. To do so you can replace `"localhost"` with the default route `"0.0.0.0"`. This will allow incoming http requests from any IP address.
+
+However, for security reasons it is recommended to use a WSGI server instead. This can be achieved by:
+
+1. installing waitress `pip install waitress`
+2. replacing the last lines of `server.py` as follows: 
+```python
+if __name__ == "__main__":
+    before_first_request()
+    from waitress import serve
+    serve(app, host='0.0.0.0', port=5000)
```

Comment:
Could we just use `waitress` by default?

Reply:
sure, ill also move the import to the top of the file

Reply:
its probably not a good idea, heres why:

a) the development server is more lightweight and better suited for a first impression & for adjusting the server to individual needs, and should definitely be mentioned inside this documentation - ill explain this in more detail within the documentation
b) ive just noticed, that here were using port=5000, however, for remote communication, Port 80 should be used - ill add an appropriate comment explaining why
c) using Port 80 requires the user to configure the firewall accordingly & also execute the script using the administrative shell. I think doing so is better suited *after* having given the user to get a first impression using the development stage server